torch/csrc/jit/passes/requires_grad_analysis.cpp

```diff
@@ -64,6 +65,17 @@ void PropagateRequiresGradSimpleNode(Node* node) {
     return setRequiresGrad(node->output(), node->input(0)->requires_grad());
   } else if (node->matches("aten::detach(Tensor self) -> Tensor")) {
     return setRequiresGrad(node->output(), false);
+  } else if (node->kind() == aten::tensor) {
+    if (auto grad_index =
+            node->schema().argumentIndexWithName("requires_grad")) {
+      if (auto const_arg = constant_as<bool>(node->inputs().at(*grad_index))) {
+        return setRequiresGrad(node->output(), *const_arg);
+      }
+    }
+    if (auto type = node->output()->type()->cast<DimensionedTensorType>()) {
+      setRequiresGrad(node->output(), at::isFloatingType(type->scalarType()));
```

Comment:
Can you not have a float tensor `DimensionedTensorType` that doesn't require grad?

Reply:
You can, but requires grad analysis is conservative and sets every tensor which we're unsure of to set require grad